core/src/test/java/org/testcontainers/containers/ReusabilityUnitTests.java

```diff
@@ -0,0 +1,200 @@
+package org.testcontainers.containers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.*;
+import com.github.dockerjava.api.model.Container;
+import com.github.dockerjava.core.command.CreateContainerCmdImpl;
+import com.github.dockerjava.core.command.InspectContainerCmdImpl;
+import com.github.dockerjava.core.command.ListContainersCmdImpl;
+import com.github.dockerjava.core.command.StartContainerCmdImpl;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.Parameterized;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+import org.rnorth.visibleassertions.VisibleAssertions;
+import org.testcontainers.containers.startupcheck.StartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@RunWith(Enclosed.class)
+public class ReusabilityUnitTests {
```

Comment:
Some IntelliJ hints for refactoring into expression lambdas in this class.

Reply:
I don't see any. Which IntelliJ IDEA are you using? I have CE, maybe that's why :)
Can you share some examples of refactorings it suggests?

Reply:
Oh, sure!
Let me try in comment.